table_reconstruction/utils/cell_utils.py

```diff
@@ -0,0 +1,262 @@
+from typing import List, Tuple
+from shapely.geometry import LineString
+from .lines_utils import is_line
+import numpy as np
+
+
+def get_intersection_points(
+    horizontal_lines: List[List],
+    vertical_lines: List[List],
+    tab_coord: List
+) -> Tuple[np.ndarray, np.ndarray]:
+    """This a function which find the coordinate (x, y) of intersection points
+
+    Args:
+        horizontal_lines (List[List]): The coordinate of horizontal lines
+        vertical_lines (List[List]): The coordinate of vertical lines
+        tab_coord (List): The coordinate of table
+
+    Returns:
+        Tuple[np.ndarray, np.ndarray]: intersection and fake intersection points
+    """
+
+    intersect_points = []
+    fake_intersect_points = []
+    tab_x1, tab_y1, tab_x2, tab_y2 = tab_coord
+
+    for h_coor in horizontal_lines:
+        h_x1, h_y1, h_x2, h_y2 = h_coor
+        fake_h_x1 = min(h_x1, tab_x1)
+        fake_h_x2 = max(h_x2, tab_x2)
+
+        h_line = LineString([(h_x1, h_y1), (h_x2, h_y2)])
+        fake_h_line = LineString([(fake_h_x1, h_y1), (fake_h_x2, h_y2)])
+
+        for v_coor in vertical_lines:
+            v_x1, v_y1, v_x2, v_y2 = v_coor
+            fake_v_y1 = min(v_y1, tab_y1)
+            fake_v_y2 = max(v_y2, tab_y2)
+
+            fake_v_line = LineString([(v_x1, fake_v_y1), (v_x2, fake_v_y2)])
+            v_line = LineString([(v_x1, v_y1), (v_x2, v_y2)])
+
+            intersect_point = h_line.intersection(v_line)
+            fake_intersect_point = fake_h_line.intersection(fake_v_line)
+
+            if len(list(intersect_point.coords)) != 0:
+                intersect_points.append(list(intersect_point.coords))
+
+            if len(list(fake_intersect_point.coords)) != 0:
+                fake_intersect_points.append(list(fake_intersect_point.coords))
+
+    final_intersect_points = np.array(intersect_points)
+    final_intersect_points = np.squeeze(final_intersect_points, axis=1)
+    final_fake_intersect_points = np.array(fake_intersect_points)
+    final_fake_intersect_points = np.squeeze(final_fake_intersect_points, axis=1)
+
+    return final_intersect_points, final_fake_intersect_points
+
+
+def is_cell_existed(
+    cell_coord: List[List],
+    thresh: float,
```

Comment:
Chưa có có phần đề cập đến giá trị `thresh` nên để là bao nhiêu
```suggestion
    thresh: float=15,
```